libact/base/dataset.py

```diff
@@ -47,17 +55,34 @@ def __len__(self):
         -------
         n_samples : int
         """
-        return len(self.data)
+        return self._X.shape[0]
+    
+    def __getitem__(self, idx):
+        # still provide the interface to direct access the data by index
+        return self._X[idx], self._y[idx]
+    
+    @property
+    def data(self): return self
+
+    def get_labeled_mask(self):
+        """
+        Get the mask of labeled entries.
+
+        Returns
+        -------
+        mask: numpy array of bool, shape = (n_sample, )
+        """
+        return ~np.fromiter( ( e is None for e in self._y), dtype=bool )
 
-    def len_labeled(self):
+    def len_labeled(self): 
```

Comment:
Is this a redundant whitespace?

Reply:
Yep, sorry I missed that. 
Let me do an update.